examples/rust/hash-and-merkle/README.md

```diff
@@ -29,27 +40,51 @@ keccak_hasher.hash(
 
 // Hash Babybear field elements
 let input_field_elements = BabybearCfg::generate_random(128);
-let mut output = vec![0u8; 32]; // Output buffer for hash result
+let mut output = vec![0u8; 1*32]; // Output buffer for batch=1
 keccak_hasher.hash(
     HostSlice::from_slice(&input_field_elements),
     &HashConfig::default(),
     HostSlice::from_mut_slice(&mut output),
 ).unwrap();
 ```
 
-This example demonstrates how to hash both simple strings and field elements using the Keccak-256 hashing algorithm, which is commonly used in cryptographic applications.
+### 2. Merkle Tree Example
 
-## What's in this example
+The Merkle tree example demonstrates **building a binary Merkle tree**, committing to string data, and generating **Merkle proofs**.
+
+```rust
+fn merkle_tree_example() {
+    let input_string = String::from(
+        "Hello, this is an ICICLE example to commit to a string and open specific parts +Add optional Pad",
+    );
+    let input = input_string.as_bytes();
+    
+    // define tree with Keccak-256 and blake2s hashers
+    let hasher = Keccak256::new(leaf_size).unwrap();
+    let compress = Blake2s::new(hasher.output_size() * 2).unwrap();    
+    let layer_hashes: Vec<&Hasher> = std::iter::once(&hasher)
+        .chain(std::iter::repeat(&compress).take(tree_height))
+        .collect();
+    let merkle_tree = MerkleTree::new(&layer_hashes, leaf_size, 0).unwrap();
+
+    // 1. commit
+    merkle_tree.build(HostSlice::from_slice(&input), &config).unwrap();
+    let commitment = merkle_tree.get_root().unwrap();
+    // 2. open leaf #3
+    let merkle_proof = merkle_tree
+        .get_proof(HostSlice::from_slice(&input), 3, true /*=pruned path*/, &config)
+        .unwrap();
+    // 3. verify
+    let proof_is_valid = merkle_tree.verify(&merkle_proof).unwrap();
```

Comment:
for the completness of the application, I think that the verifier needs to create a new instance of a merkle tree and use the same instance used for the build

Reply:
done